src/Castle.Windsor/MicroKernel/DefaultKernel.cs

```diff
@@ -342,7 +342,17 @@ public virtual void AddSubSystem(String name, ISubSystem subsystem)
 			}
 			else if (name == SubSystemConstants.NamingKey)
 			{
-				NamingSubSystem = (INamingSubSystem)subsystem;
+				var namingSubSystem = subsystem as INamingSubSystem;
+
+				if (NamingSubSystem != null)
+				{
+					foreach (var handler in NamingSubSystem.GetAllHandlers())
+					{
+						namingSubSystem?.Register(handler);
```

Comment:
@robertcoltheart we cannot use c notation in Castle code standards, you need to if(namingSubSystem!=null)